File: numberDisappeared.js
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


var findDisappearedNumbers = function(nums) {
    let result = [];
    if(nums == null || nums.length == 0)return result;
    
    for(let i = 0; i < nums.length; i++){
        let idx = Math.abs(nums[i]) - 1;
        if(nums[idx] > 0){
            nums[idx] *= -1
        }
    }
    for(let i = 0; i < nums.length; i++){
        if(nums[i] > 0){
            result.push(i+1)
        }
        else {
            nums[i] *= -1
        }
    }
    return result
    
};

File: numberDisappeared.java
import java.util.ArrayList;
import java.util.List;

class Solution_ {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int max=0;
        int[] data = new int[nums.length+1];
        List<Integer> result = new ArrayList<Integer>();
        for(int i=0;i<nums.length;i++){
            data[nums[i]]=nums[i];

        }
        for (int i=0;i<data.length;i++){
            if(data[i]!=i){
                result.add(i);
            }
        }

        return result;
    }
}